src/core/unif.ml

```diff
@@ -61,19 +60,24 @@ let nl_distinct_vars : term array -> tvar array option =
   try Some (Array.map replace_nl_var (Array.map to_var ts))
   with Not_a_var -> None
 
-(** [instantiate m ts u] check whether [m] can be instantiated for solving the
-    unification problem “m[ts] = u”. The returned boolean tells whether [m]
-    was instantiated or not. *)
+(** [instantiation m ts u] checks whether, in a problem [m[ts]=u], [m] can be
+   instantiated and returns the corresponding instantiation. It does not check
+   whether the instantiation is closed though. *)
+let instantiation : meta -> term array -> term ->
+                    (term, term) Bindlib.mbinder Bindlib.box option
```

Comment:
You can use type `tbinder` (exported by `Terms`).

Reply:
No, its a mbinder I need.